### PR TITLE
Added TestCase for GaussianDistribution::Train

### PR DESCRIPTION
**Pull request by Lakshya1605**
#839

### CODE REVIEW

File: src/mlpack/tests/distribution_test.cpp [hunk 1]
```diff
@@ -390,10 +390,145 @@ BOOST_AUTO_TEST_CASE(GaussianDistributionTrainTest)
...
+{
+  double mean = 5.0;
+  double stddeviation = 2.0;
+
+  //Creates a normal distribution generator.
+  std::default_random_engine generator;
+  generator.seed(std::time(NULL));
+  std::normal_distribution<double> dist(mean, stddeviation);
+  
+  size_t N = 50000;
```
Comment (rcurtin) [inline src/mlpack/tests/distribution_test.cpp]: I think only 5k samples should be fine for this test---no need to use more samples than are needed.

File: src/mlpack/tests/distribution_test.cpp [hunk 2]
```diff
@@ -390,10 +390,145 @@ BOOST_AUTO_TEST_CASE(GaussianDistributionTrainTest)
...
+**/
+BOOST_AUTO_TEST_CASE(GaussianDistributionTrainWithProbabilitiesTest)
+{
+  double mean = 5.0;
+  double stddeviation = 2.0;
+
+  //Creates a normal distribution generator.
+  std::default_random_engine generator;
+  generator.seed(std::time(NULL));
+  std::normal_distribution<double> dist(mean, stddeviation);
```
Comment (rcurtin) [inline src/mlpack/tests/distribution_test.cpp]: Please don't set the random seed for anything other than testing.  It's important that failing tests be reproducible, and if the failure depends on the random seed, and the seed is set in the test, then it's not easily reproducible.

File: src/mlpack/tests/distribution_test.cpp [hunk 3]
```diff
@@ -390,10 +390,145 @@ BOOST_AUTO_TEST_CASE(GaussianDistributionTrainTest)
...
+  
+  size_t N = 50000;
+  size_t d =1;
+  
+  arma::mat rdata(d, N);
+  
+  for (size_t i = 0; i < d; i++)
+    for (size_t j = 0; j < N; j++)
+      rdata(i,j) = dist(generator);
```
Comment (rcurtin) [inline src/mlpack/tests/distribution_test.cpp]: You can just use `GaussianDistribution::Random()` here.

File: src/mlpack/tests/distribution_test.cpp [hunk 4]
```diff
@@ -390,10 +390,145 @@ BOOST_AUTO_TEST_CASE(GaussianDistributionTrainTest)
...
+
+  //Creates a uniform distribution generator
+  std::uniform_real_distribution<double> prob(0, 1);
+  arma::vec probabilities(N);
+  
+  for (size_t i = 0; i < N; i++)
+    probabilities(i) = prob(generator);
```
Comment (rcurtin) [inline src/mlpack/tests/distribution_test.cpp]: You can just use `math::Random()` here.

File: src/mlpack/tests/distribution_test.cpp [hunk 5]
```diff
@@ -390,10 +390,145 @@ BOOST_AUTO_TEST_CASE(GaussianDistributionTrainTest)
...
+  //fits data with only data
+  GaussianDistribution guDist;
+  guDist.Train(rdata);
+
+  //fits result with data and each probability as 1
+  GaussianDistribution guDist2;
+  guDist2.Train(rdata, probabilities);
+
+  BOOST_REQUIRE_CLOSE(guDist.Mean()[0], guDist2.Mean()[0], 1e-15);
+  BOOST_REQUIRE_CLOSE(guDist.Covariance()[0], guDist2.Covariance()[0], 1e-2);
```
Comment (rcurtin) [inline src/mlpack/tests/distribution_test.cpp]: I looked into why a larger tolerance is necessary here.  It has to do with how the covariance is normalized in the case with and without probabilities.  So 1e-2 is fine.

**Comment by Lakshya1605** [comment]
@rcurtin  I have modified the code as specified .I have also tried to see to that my code matches with the Style Guide ,if something is wrong please let me know.
Thank You !!!

**Comment by rcurtin** [comment]
Ok, looks better.  There are still some style issues but I will fix them after merge.  Would you like to add your name to the list of contributors in `src/mlpack/core.hpp` and `COPYRIGHT.txt`?

**Comment by Lakshya1605** [comment]
Yes i would love to add my name.Thank You !!!

**Pull request merged**